pkg/kubectl/scheme/install.go

```diff
@@ -17,288 +17,46 @@ limitations under the License.
 package scheme
 
 import (
-	admissionv1alpha1 "k8s.io/api/admission/v1alpha1"
-	admissionregistrationv1alpha1 "k8s.io/api/admissionregistration/v1alpha1"
-	appsv1 "k8s.io/api/apps/v1"
-	appsv1beta1 "k8s.io/api/apps/v1beta1"
-	appsv1beta2 "k8s.io/api/apps/v1beta2"
-	authenticationv1 "k8s.io/api/authentication/v1"
-	authenticationv1beta1 "k8s.io/api/authentication/v1beta1"
-	authorizationv1 "k8s.io/api/authorization/v1"
-	authorizationv1beta1 "k8s.io/api/authorization/v1beta1"
-	autoscalingv1 "k8s.io/api/autoscaling/v1"
-	autoscalingv2beta1 "k8s.io/api/autoscaling/v2beta1"
-	batchv1 "k8s.io/api/batch/v1"
-	batchv1beta1 "k8s.io/api/batch/v1beta1"
-	batchv2alpha1 "k8s.io/api/batch/v2alpha1"
-	certificatesv1beta1 "k8s.io/api/certificates/v1beta1"
-	corev1 "k8s.io/api/core/v1"
-	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
-	imagepolicyv1alpha1 "k8s.io/api/imagepolicy/v1alpha1"
-	networkingv1 "k8s.io/api/networking/v1"
-	policyv1beta1 "k8s.io/api/policy/v1beta1"
-	rbacv1 "k8s.io/api/rbac/v1"
-	rbacv1alpha1 "k8s.io/api/rbac/v1alpha1"
-	rbacv1beta1 "k8s.io/api/rbac/v1beta1"
-	schedulingv1alpha1 "k8s.io/api/scheduling/v1alpha1"
-	settingsv1alpha1 "k8s.io/api/settings/v1alpha1"
-	storagev1 "k8s.io/api/storage/v1"
-	storagev1beta1 "k8s.io/api/storage/v1beta1"
-	"k8s.io/apimachinery/pkg/apimachinery/announced"
 	"k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/apimachinery/pkg/util/sets"
-	"k8s.io/client-go/kubernetes/scheme"
+
+	admission "k8s.io/api/admission/install"
+	admissionregistration "k8s.io/api/admissionregistration/install"
+	apps "k8s.io/api/apps/install"
+	authentication "k8s.io/api/authentication/install"
+	authorization "k8s.io/api/authorization/install"
+	autoscaling "k8s.io/api/autoscaling/install"
+	batch "k8s.io/api/batch/install"
+	certificates "k8s.io/api/certificates/install"
+	core "k8s.io/api/core/install"
+	extensions "k8s.io/api/extensions/install"
+	imagepolicy "k8s.io/api/imagepolicy/install"
+	networking "k8s.io/api/networking/install"
+	policy "k8s.io/api/policy/install"
+	rbac "k8s.io/api/rbac/install"
+	scheduling "k8s.io/api/scheduling/install"
+	settings "k8s.io/api/settings/install"
+	storage "k8s.io/api/storage/install"
 )
 
-// Register all groups in the kubectl's registry, but no componentconfig group since it's not in k8s.io/api
-// The code in this file mostly duplicate the install under k8s.io/kubernetes/pkg/api and k8s.io/kubernetes/pkg/apis,
-// but does NOT register the internal types.
 func init() {
-	// Register external types for Scheme
 	v1.AddToGroupVersion(Scheme, schema.GroupVersion{Version: "v1"})
-	scheme.AddToScheme(Scheme)
```

Comment:
Why Deleting this? Does kubectl need to register non-core groups apis for scheme?

Reply:
This is what all the install calls are for. Compare the internal client: https://github.com/kubernetes/kubernetes/blob/master/pkg/client/clientset_generated/internalclientset/scheme/register.go#L57